autoscale_cloudroast/test_repo/autoscale/system/integration/test_system_integration_rcv3.py

```diff
@@ -0,0 +1,757 @@
+                          msg='The create failed with {0} for group '
+                          '{1}'.format(pool_group_resp.status_code, pool_group.id))
+        self.assertEquals(pool_group.launchConfiguration.loadBalancers[0].loadBalancerId,
+                          self.pool_1.id,
+                          msg='The launchConfig for group {0} did not contain the load balancer'
+                          .format(pool_group.id))
+        self.assertEquals(pool_group.launchConfiguration.loadBalancers[0].type, "RackConnectV3",
+                          msg='Load balancer type {0} is not correct for RackConnect pools'
+                          .format(pool_group.launchConfiguration.loadBalancers[0].type))
+
+    #@unittest.skip("Skipping")
+    @tags(speed='slow', type='rcv3')
+    def test_create_scaling_group_with_pool_on_public(self):
+        """
+        Test that it is possible to create a scaling group with 0 entities
+        connected to an RCV3 LB pool with only private network specified
+        """
+        # Create a scaling group with zero servers
+        lb_pools = [{'loadBalancerId': self.pool_1.id, 'type': 'RackConnectV3'}]
+        pool_group_resp = self._create_rcv3_group(lb_list=lb_pools,
+                                                  group_min=0,
+                                                  network_list=[self.public_network])
+
+        pool_group = pool_group_resp.entity
+
+        self.assertTrue(pool_group_resp.ok,
+                        msg='Create scaling group call failed with API Response: {0} for '
+                        'group {1}'.format(pool_group_resp.content, pool_group.id))
+        self.assertEquals(self.create_resp.status_code, 201,
+                          msg='The create failed with {0} for group '
+                          '{1}'.format(pool_group_resp.status_code, pool_group.id))
+        self.assertEquals(pool_group.launchConfiguration.loadBalancers[0].loadBalancerId,
+                          self.pool_1.id,
+                          msg='The launchConfig for group {0} did not contain the load balancer'
+                          .format(pool_group.id))
+        self.assertEquals(pool_group.launchConfiguration.loadBalancers[0].type, "RackConnectV3",
+                          msg='Load balancer type {0} is not correct for RackConnect pools'
+                          .format(pool_group.launchConfiguration.loadBalancers[0].type))
+
+    #@unittest.skip("Skipping")
+    @tags(speed='slow', type='rcv3')
+    def test_create_scaling_group_with_pool_counts(self):
+        """
+        Test that it is possible to create a scaling group with min_servers servers
+        connected to an RCv3 LB pool.
+
+        This is a simple smoke test that only checks the node counts.
+        """
+        # This sleep is necessary because other tests have just completed, but
+        # their resources haven't yet been completely freed up.  TODO(sfalvo):
+        # See Github issue #855.
```

Comment:
I'm not sure how I can communicate this more clearly.

What I'm saying we need is to add polling to delete_X, for whatever X we care to implement.  X doesn't have to be scaling group, and in all honesty, it's rarely the cause of our intermittent failures.  However, to use it as an example, if delete_scaling_group can be altered so that it lists its resources prior to actually issuing the delete, then we have plenty of information on-hand to poll to make sure that all its attached resources are properly disposed of.  OR, if we detect an error in the process, we can fail an assert.

It's a lot better than just waiting and hoping for the best.